src/io/flutter/view/preview/DartElementPresentationUtil.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.view.preview;
+
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.ui.LayeredIcon;
+import com.jetbrains.lang.dart.DartComponentType;
+import com.jetbrains.lang.dart.util.DartPresentableUtil;
+import org.dartlang.analysis.server.protocol.Element;
+import org.dartlang.analysis.server.protocol.ElementKind;
+
+import javax.swing.*;
+
+import static com.intellij.icons.AllIcons.Nodes.*;
+import static com.intellij.icons.AllIcons.Nodes.PropertyWrite;
+
+public class DartElementPresentationUtil {
+  private static final LayeredIcon STATIC_FINAL_FIELD_ICON = new LayeredIcon(Field, StaticMark, FinalMark);
+  private static final LayeredIcon FINAL_FIELD_ICON = new LayeredIcon(Field, FinalMark);
+  private static final LayeredIcon STATIC_FIELD_ICON = new LayeredIcon(Field, StaticMark);
+  private static final LayeredIcon STATIC_METHOD_ICON = new LayeredIcon(Method, StaticMark);
+  private static final LayeredIcon TOP_LEVEL_FUNCTION_ICON = new LayeredIcon(Function, StaticMark);
+  private static final LayeredIcon TOP_LEVEL_VAR_ICON = new LayeredIcon(Variable, StaticMark);
+  private static final LayeredIcon CONSTRUCTOR_INVOCATION_ICON = new LayeredIcon(Class, TabPin);
+  private static final LayeredIcon FUNCTION_INVOCATION_ICON = new LayeredIcon(Method, TabPin);
+  private static final LayeredIcon TOP_LEVEL_CONST_ICON = new LayeredIcon(Variable, StaticMark, FinalMark);
+
+  public static Icon getIcon(Element element) {
```

Comment:
Is this from the Dar plugin, and if so, can you add a link to the file on github (to help us better keep them in sync)?

Reply:
I added the link to the class documentation.